SOT-176: Add Verifiable Secret Sharing doc (#79)
sota-zk-labs/zkp-documents: terms/verifiable_secret_sharing.md

---
comments: true
---
# Verifiable Secret Sharing

## Distributed Key Generation

Read the definition of DKG [here](./distributed_key_generation.md).

## Verifiable Secret Sharing (VSS)

### Definition

> A $(t, n)$ VSS includes two phases: *Share* and *Reconstruct* as follow:

Reviewer: ```suggestion
A $(t, n)$ VSS includes two phases: *Sharing* and *Reconstruction* as follow:
```

Author: I followed the definition in Orochi cookbook:
https://docs.orochi.network/dkg/verifiable-secret-sharing/syntax.html

**Share:**

- Dealer $D$ has a secret $s$. There are $n$ parties $P_1,\dots,P_n$.
- After several rounds of interaction among the parties, each party holds a share $s_i$.

**Reconstruct**:

> - Each party $P_i$ publishes its share $s_i$ from the sharing phase for reconstruction $s$.

Reviewer: ```suggestion
- Each party $P_i$ publishes its share $s_i$ from the sharing phase for the reconstruction of $s$.
```

Reviewer: to reconstruct

Reviewer: for reconstructing

> If $D$ is dishonest, then at the end of **sharing** phase, there exist a value $s^*$ such that all parties agree on it at the end of

Reviewer: ```suggestion
If $D$ is dishonest, then at the end of **sharing** phase, there exists a value $s^*$ such that all parties agree on it at the end of
```

**reconstruct** phase.

## Pedersen Construction

We describe a VSS protocol from **Pedersen**.

### The Commitment Scheme

Let $p, q$ be large primes such that $q$ divides $p-1$ and let $G_q$ be the unique subgroup of $\mathbb{Z}^*_p$ of order $q$. Let $g$
and $h$ be elements of $G_q$ where $g$ is the generator of $G_q$ and no one knows the value $d$ such that $h = g^d$. These elements
can be chosen using a coin-flipping protocol.

> The committer commits to its secret $s \in \mathbb{Z}_q$ by choosing a random $t \in \mathbb{Z_q}$, then computing and broadcasting

Reviewer: $t$ is already used for threshold
